File: risk-openapi-php-demo/mobile/riskDetail/RiskDetailDemo.php
<?php
/** 反外挂嫌疑明细数据查询接口API示例 */
/** 产品ID，从【易盾官网-服务管理-已开通业务】页面获取 */
const APPID = "your_app_id";
/** 密钥，从【易盾官网-服务管理-已开通业务】页面获取 */
const APPKEY = "your_app_key";
/** 接口URL */
const API_URL = "http://open-yb.dun.163.com/api/open/v2/risk/detail_data/list";
/** API timeout*/
const API_TIMEOUT = 2;
require("../../util.php");

/**
 * 反垃圾请求接口简单封装
 * $params 请求参数
 */
function check($params)
{
    $params["appId"] = APPID;
    $params["timestamp"] = time() * 1000;// time in milliseconds
    $params["nonce"] = sprintf("%d", rand()); // random int
    $params["token"] = SIGNATURE_METHOD;

    $params = toUtf8($params);
    $params["token"] = gen_signature(APPKEY, APPID, $params["timestamp"], $params["nonce"]);

    $result = curl_post($params, API_URL, API_TIMEOUT);
    if ($result === FALSE) {
        return array("code" => 500, "msg" => "file_get_contents failed.");
    } else {
        return json_decode($result, true);
    }
}

// 简单测试
function main()
{
    echo "mb_internal_encoding=" . mb_internal_encoding() . "\n";

    $params = array(
        // 1：不去重查询；0：去重查询（默认）
        "duplicate" => 1,
        // 0: 使用客户端事件发生时间（默认）; 1:使用入库时间
        "queryTimeType" => 0,
        "beginDateTime" => time() * 1000,
        "endDateTime" => time() * 1000,
        // 用于分页查询的关联标记。第一次查询时，该字段填充""
        // 当使用分页查询时，startFlag字段使用上一次返回值填充
        // params.put("startFlag", "");
        // 0: 返回数据格式为LinedText格式; 1: 返回数据格式为JSON格式; 默认为0
        "formatType" => 1
    );
    var_dump($params);

    $ret = check($params);
    var_dump($ret);
    if ($ret["code"] == 200) {
        $data = $ret["data"];
        echo "请求成功，data: {$data}";
    } else {
        $msg = $ret["msg"];
        echo "请求失败，msg: {$msg}";
        var_dump($ret);
    }
}

main();
?>

File: risk-openapi-go-demo/mobile/riskDetail/RiskDetailDemo.go
package main

import (
	"crypto/md5"
	"encoding/hex"
	"encoding/json"
	"fmt"
	simplejson "github.com/bitly/go-simplejson"
	"io/ioutil"
	"math/rand"
	"net/http"
	"sort"
	"strconv"
	"strings"
	"time"
)

const (
	apiUrl = "http://open-yb.dun.163.com/api/open/v2/risk/detail_data/list" //接口url
	appId  = "your_app_id"                                                  //产品ID，从【易盾官网-服务管理-已开通业务】页面获取
	appKey = "your_app_key"                                                 //密钥，从【易盾官网-服务管理-已开通业务】页面获取
)

// 请求易盾接口
func check(params map[string]interface{}) *simplejson.Json {
	marshal, _ := json.Marshal(params)
	jsonStr := string(marshal)
	println(jsonStr)
	resp, err := http.Post(apiUrl, "application/json", strings.NewReader(jsonStr))

	if err != nil {
		fmt.Println("调用API接口失败:", err)
		return nil
	}

	defer resp.Body.Close()

	contents, _ := ioutil.ReadAll(resp.Body)
	result, _ := simplejson.NewJson(contents)
	return result
}

// 生成签名信息
func genSignature(appKey string, appId string, timestamp string, nonce string) string {
	var params map[string]string
	params = make(map[string]string)
	params["appId"] = appId
	params["timestamp"] = timestamp
	params["nonce"] = nonce
	var paramStr string
	keys := make([]string, 0, len(params))
	for k := range params {
		keys = append(keys, k)
	}
	sort.Strings(keys)
	for _, key := range keys {
		paramStr += key + params[key]
	}
	paramStr += appKey
	md5Reader := md5.New()
	md5Reader.Write([]byte(paramStr))
	return hex.EncodeToString(md5Reader.Sum(nil))
}

func main() {
	var timestamp = strconv.FormatInt(time.Now().UnixNano()/1000000, 10)
	var nonce = strconv.FormatInt(rand.New(rand.NewSource(time.Now().UnixNano())).Int63n(10000000000), 10)
	var token = genSignature(appKey, appId, timestamp, nonce)
	params := map[string]interface{}{
		"appId":         appId,
		"timestamp":     timestamp,
		"nonce":         nonce,
		"token":         token,
		"duplicate":     "1",
		"queryTimeType": "0",
		"beginDateTime": timestamp,
		"endDateTime":   timestamp,
		"formatType":    "1",
		"roleId":        "",
		"roleIdList":    strings.Join([]string{"roleId1"}, ","),
		"account":       "",
		"accountList":   strings.Join([]string{"acount1"}, ","),
		"ip":            "",
		"ipList":        strings.Join([]string{"ip1"}, ","),
		"packageName":   "",
		"appVersion":    "",
		"thirdLevelTagName":    "1",
		"thirdLevelTagNameList":       strings.Join([]string{"ip1"}, ","),
		"dataType":      "1"


	}

	ret := check(params)

	code, _ := ret.Get("code").Int()
	message, _ := ret.Get("msg").String()
	data, _ := ret.Get("data").MarshalJSON()
	if code == 200 {
		fmt.Printf("请求成功：msg=%s, data=%s", message, data)
	} else {
		fmt.Printf("请求失败: code=%d, msg=%s", code, message)
	}
}
